datastream/dataset.py

```diff
@@ -125,12 +126,31 @@ def map(
         ... )[-1]
         4
         '''
+
+        def composed_fn(dataframe, index):
+            item = self.get_item(dataframe, index)
+            try:
+                return function(item)
+            except Exception as e:
+                item_text = str(item)
```

Comment:
Can we always guarantee that `item` implements `__str__`?

Reply:
Or I guess it doesn't matter since `str` can never crash, right?

Reply:
Not sure. I think we should test this a bit more

Reply:
a general guideline
```
    The __repr__ is for developers. The essential purpose is to explain what the object is in an unambiguous way.

    The __str__ is for end users. The essential purpose is to explain what data the object contains in a readable manner.
```
`repr(.)` always exist (unless you explicitly remove it)